src/index.tsx

```diff
@@ -10,7 +12,9 @@ import { register } from './serviceWorker';
 const render = () => {
   ReactDOM.render(
     <AppContainer>
-      <App />
+      <GlobalProvider>
+        <App />
+      </GlobalProvider>
```

Comment:
Same, you can write it here instead. Seems like an over optimisation to have a render component function inside your context file seems it only works for this case. I'd write it here or use a HOC instead